### ISSUE TITLE
Allow ignore to be on any line of multi-line statement

### ISSUE THREAD

**Issue by Zeckie** [python/mypy#12341]
**Feature**

Currently, comments can be added to a line to ignore either all errors (`# type: ignore`), or a specific error code (eg. `# type: ignore[call-arg]`). When a single statement spans multiple lines, the comment has to be added on the line that mypy "decides" is the one with the error (is it the first line of the expression, rather than first line of the statement?). In some cases, it would make sense to allow the comment to be on a different line of the same statement. I propose allowing it to be on any line of the statement, and therefore to apply to the whole statement.

**Pitch**
Sometimes it makes more sense for it to be another line, such as when importing something that mypy thinks does not exist, it makes sense to have the ignore on the same line as the attribute that is reported as not existing.
```
from typing import ( # ignore currently needs to be here
    Union,
    DoesNotExist, # type: ignore[error attr-defined]
)
```

Sometimes, a formatter like black changes a long line into a multiple line statement. The comment that was at the end of the line is moved to the end of the last line. 
```
def foo(bar:int) -> str: ...
def baz() -> int: ...

a = ( # type: ignore[call-arg]
    "something" 
    if baz() > 0 else 
    foo() # ignore currently needs to be here
)  # formatters (eg. black) often put ignore here
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-codes%2Cstrict%2Cwarn-unused-ignores&gist=e41c530d401ac4f21cb7a21e68fea141

**Comment by DetachHead** [comment]
i agree with the first example because you should be able to split an expression up such that the part with the error is isolated on a single line, to minimize the chance of the `type:ignore` comment effecting other parts of the expression too

in the second example i think it's preferable for it to require the `type:ignore` comment to be on the line with the error, because otherwise it's easy to accidentally insert a new error and ignore it:

```py
def foo(bar:int) -> str: ...
def baz(bar: int) -> int: ... # new change, `baz` now requires an arg

a = ( # type: ignore[call-arg]
    "something" 
    if baz() > 0 else 
    foo()
)  # formatters (eg. black) often put ignore here
```
if it allowed the comment on any line in the statement, now we get no new error. the `type:ignore` that was intended to only suppress the error on `foo` now also suppresses the new error as well

**Comment by llucax** [comment]
Another use case is linters that check for maximum line length. For example (real example):

```py
def wall_now() -> datetime:
    """Get the current wall clock time from the mocked datetime in the target module."""
    # Disable isort formatting because it wants to put the ignore in the wrong line,
    #  we now also have to ignore the maximum line length (E501)
    # isort: off
    # pylint: disable-next=import-outside-toplevel
    from frequenz.sdk.timeseries._resampling._wall_clock_timer import (  # type: ignore[attr-defined] # noqa: E501
        datetime as mock_datetime,
    )

    # isort: on

    return mock_datetime.now(timezone.utc)
```

I couldn't find a better way to put this. With multi-line (or next-line) ignore it could just be:

```py
def wall_now() -> datetime:
    """Get the current wall clock time from the mocked datetime in the target module."""
    # pylint: disable-next=import-outside-toplevel  # type: ignore-next[attr-defined]
    from frequenz.sdk.timeseries._resampling._wall_clock_timer import (
        datetime as mock_datetime,
    )

    return mock_datetime.now(timezone.utc)
```
